docs/features/network-security-controls/admin-network-policy.md

```diff
@@ -1,303 +1,1118 @@
-* Since we can have upto 100 policies and each one can have upto 100 gress rules (100*100), we have blocked out the range: 30,000 - 20,000 priority range for the OVN nbdb.ACL table in the `Tier1` block for ANP's implementation.
-* Each AdminNetworkPolicy CRD will have a subject on which the policy is applied on - this is translated to one PortGroup on which the ACLs of each rules are attached on.
-* Each gress rule can have upto 100 peers. Each rule will also create an nbdb.AddressSet which will contain the IPs of all the pods that are selected by the peer selector across all the peers of that given rule.
+#### BaselineAdminNetworkPolicy Sample API
 
-The PortGroup for the above AdminNetworkPolicy is:
+Unlike ANP's BANP is a singleton object in your
+cluster meant to create a guardrail default security
+posture.
 
+Let us look at a sample BANP yaml:
+
+```yaml
+apiVersion: policy.networking.k8s.io/v1alpha1
+kind: BaselineAdminNetworkPolicy
+metadata:
+  name: default
+spec:
+  subject:
+    namespaces: {}
+  ingress:
+  - name: "default-deny"
+    action: "Deny"
+    from:
+    - namespaces:
+        namespaceSelector: {}
+  egress:
+  - name: "default-deny"
+    action: "Deny"
+    to:
+    - namespaces:
+        namespaceSelector: {}
 ```
-_uuid               : a10e3675-5260-4e28-9462-b705b9dac862
-acls                : [120082fa-5a70-4c72-9211-529766078278, 2e0f811f-e0db-41ad-b12b-4a0cf1c621ae]
-external_ids        : {AdminNetworkPolicy=pass-example}
-name                : a3052488126344707991
-ports               : [a22a4c3a-bb65-4b22-8bc1-13e1e8899a7b, c7e4ffe3-73df-4db5-a3bc-a9649394d549]
+
+You can do `oc apply -f default.yaml` to create this ANP in your cluster
+```shell
+$ oc get banp
+NAME      AGE
+default   16m
 ```
 
-The ACLs for the above AdminNetworkPolicy are:
+### OVN-Kubernetes Implementation Details
 
-```
-_uuid               : 2e0f811f-e0db-41ad-b12b-4a0cf1c621ae
+We have a new level driven controller in OVN-Kubernetes
+called [admin_network_policy](https://github.com/ovn-org/ovn-kubernetes/tree/master/go-controller/pkg/ovn/controller/admin_network_policy)
+which watches for the following objects:
+
+* AdminNetworkPolicy
+* BaselineAdminNetworkPolicy
+* Namespaces
+* Pods
+* Nodes
+
+in your cluster. For each add, update and delete event
+that this controller cares about for each of the above
+objects, it add's it to the centralized ANP and BANP
+queue in a single threaded fashion.
+
+The controller maintains an internal cache state for each
+corresponding ANP and tries to level the current state
+configuration for a given ANP with the desired state
+configuration for that ANP.
+
+It also emits events like `ANPWithUnsupportedPriority` and
+`ANPWithDuplicatePriority` so that users are aware in those
+circumstances.
+
+#### Pass Action: Delegate decision to NetworkPolicies
+
+In addition to setting `Deny` and `Allow` actions on ANP API rules,
+one can also use the `Pass` action for a rule. What this means is
```

Comment:
(Note that you already used `Pass` in the examples before, without really explaining it.)

Reply:
good point, I have left a comment on the above pass rules bit earlier on the pass rules (Please check the following sections to learn more about what `pass` is.)